src/App.tsx

```diff
@@ -1,18 +1,26 @@
-import React from 'react';
+import React, { useState } from 'react';
+import { Pagination } from './components/Pagination';
 import './App.css';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const items = getNumbers(1, 42)
   .map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [selectedValue, setSelectedValue] = useState('5');
+  const [active, setActive] = useState(1);
+  const optionValue = ['3', '5', '10', '20'];
+  const sliceItems = items.slice(
+    +selectedValue * active - +selectedValue, +selectedValue * active,
+  );
```

Comment:
use useMemo()

Reply:
this is the next topic of study, at this stage I do not have knowledge on this topic